readthedocs/rtd_tests/tests/test_doc_building.py

```diff
@@ -1610,9 +1614,10 @@ def test_is_obsolete_with_json_same_data_as_version(self, load_config):
             build_env=self.build_env,
             config=config,
         )
-        env_json_data = '{"build": {"image": "readthedocs/build:2.0", "hash": "a1b2c3"}, "python": {"version": 3.5}}'  # noqa
-        with patch('os.path.exists') as exists, patch('readthedocs.doc_builder.python_environments.open', mock_open(read_data=env_json_data)) as _open:  # noqa
+        env_json_data = '{"build": {"image": "readthedocs/build:2.0", "hash": "a1b2c3"}, "python": {"version": 3.5}, "envvars_hash": 1234}'  # noqa
+        with patch('os.path.exists') as exists, patch('readthedocs.doc_builder.python_environments.PythonEnvironment._get_envvars_hash') as get_hash, patch('readthedocs.doc_builder.python_environments.open', mock_open(read_data=env_json_data)) as _open:  # noqa
```

Comment:
Don't mock the `_get_envvar_hash` here.

Create the proper environment variables in the testing project. This should work out of the box.

Reply:
Thanks.
I have pushed the changes.